src/flint/fmpq_poly.jl

```diff
@@ -635,6 +635,56 @@ function gcdx(x::fmpq_poly, y::fmpq_poly)
    return (z, u, v)
 end
 
+###############################################################################
+#
+#   Square root
+#
+###############################################################################
+
+function sqrt(x::fmpq_poly; check::Bool=true)
+   R = parent(x)
+   d = denominator(x)
+   sd = sqrt(d; check=check)
+   n = polynomial(ZZ, [])
+   ccall((:fmpq_poly_get_numerator, libflint), Nothing,
```

Comment:
This is also fine, but by doing it in julia we are making an unnecessary copy of the input data (this is not a serious objection).